### PR TITLE
Removes unused parameter definitions

### PR DESCRIPTION
**Pull request by HendersonSC**
This just cleans up some unused code, namely parameters which are defined but unused.

### CODE REVIEW

File: src/MatrixTypes.f90 [hunk 1]
```diff
@@ -803,7 +803,6 @@ PURE SUBROUTINE trsv_sparse(uplo,trans,diag,a,ia,ja,x,incx_in)
 !> @param beta the scalar used to scale @c C
 !>
     SUBROUTINE matmult_MatrixType(A,B,C,alpha,beta,transA,transB)
-      CHARACTER(LEN=*),PARAMETER :: myName='matmult_MatrixType'
```
Comment (aarograh) [inline src/MatrixTypes.f90]: Really could just move this one down inside the preproc symbol instead

**Comment by aarograh** [comment]
I'm good with these if @stimpsonsg can merge

**Comment by stimpsonsg** [comment]
hmmm, we might need unused var, private module var, parameter, etc. checks in this CI too?

**Comment by aarograh** [comment]
yeah, I think so.  Shane and I just talked about some tweaks to how we do that check in the MPACT CI.  Once that's nailed down, we can do the same for this one too.  Probably good as a separate merge request though.

**Pull request merged**